platforms/core-configuration/configuration-problems-base/src/main/kotlin/org/gradle/internal/cc/impl/problems/JsonModelWriterCommon.kt

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.cc.impl.problems
+
+import org.apache.groovy.json.internal.CharBuf
+import java.io.Writer
+
+open class JsonModelWriterCommon(val writer: Writer) {
+    // This is a map of booleans that indicate whether the next element in the array should be preceded by a comma.
+    private var commaContext = false
```

Comment:
❌  The comment is outdated, and I don't one is required.

I'd also suggest to rename the field to a more telling `nextItemNeedsComma` or something along those lines. It's also not only about arrays, but properties within an object.